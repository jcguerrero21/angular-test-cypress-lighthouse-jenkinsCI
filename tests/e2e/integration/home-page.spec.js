/// <reference types="Cypress" />

describe('Home Page', () => {

  beforeEach(() => {
    cy.visit('/'); // go to the home page
  });

  it('cy.title() - get the title', () => {
    cy.title().should('include', 'MyCypressApp');
  });

  it('Fill the form', () => {
    cy.get('[data-cy=nameInput]').type('mavi').should('have.value', 'mavi');

    cy.get('[data-cy=emailInput]').type('mavi@email.com').should('have.value', 'mavi@email.com');

    cy.get('#mat-radio-3').click();
    cy.get('#mat-radio-3').should('have.class', 'mat-radio-checked');

    cy.get('.mat-icon-button').click();
    cy.get('.mat-calendar-body-active').click();

    cy.get('[data-cy=select]').click();
    cy.get('.mat-option-text').contains('8th Grade').then(option => {
    option[0].click();
    // After click, mdc-select should hold the text of the selected option
    cy.get('[data-cy=select]').contains('8th Grade');
    });
    cy.get('[data-cy=subjectInput]').type('math').should('have.value', 'math');
    cy.get('[data-cy=subjectInput]').type('{downarrow}{enter}')
  }); 

  // other option @cypress/snapshot
  it('toMatchSnapshot - Body', () => {
    cy.get('body').toMatchSnapshot();
    cy.get('body').toMatchImageSnapshot();
  });

  // other option npm install --save-dev @percy/cypress -> cy.percySnapshot();
  //I try install but problem with -> Failed at puppeteer@5.0.0
  //it('percySnapshot', () => {
  // Take a snapshot for visual diffing
  //  cy.percySnapshot();
  //});

  it('click on submit button', () => {
    cy.get('[data-cy=submit]').click();
  }); 

  it('cy.reload() - reload the page', () => { 
    // reload the page without using the cache
    cy.reload(true);
  });

  it('input elements are empty', () => {
    cy.get('[data-cy=nameInput]').should('have.value', '')
    cy.get('[data-cy=emailInput]').should('have.value', '')
  });

  // specific browser
  //it('cy.screenshot() - take a screenshot 800x600 pixel image', { browser: 'firefox' }, () => {
    it('cy.screenshot() - take a screenshot 800x600 pixel image', () => {
      cy.screenshot('my-cypress-app'); //produces a 800x600 pixel image
    });

    it('cy.screenshot() - take a screenshot 1920x1080 pixel image', () => {
      cy.viewport(1920,1080);
      cy.screenshot('my-cypress-app-desktop'); 
    });

    it('cy.screenshot() - take a screenshot 360x640 pixel image', () => {
      cy.viewport(360,640);
      cy.screenshot('my-cypress-app-galaxyS5'); 
    });

    it('cy.screenshot() - take a screenshot 768x1024 pixel image', () => {
      cy.viewport(768,1024);
      cy.screenshot('my-cypress-app-ipad');
    });

    it('cy.screenshot() - take a screenshot 375x667 pixel image', () => {
      cy.viewport(375,667);
      cy.screenshot('my-cypress-app-iphone'); 
    });
});