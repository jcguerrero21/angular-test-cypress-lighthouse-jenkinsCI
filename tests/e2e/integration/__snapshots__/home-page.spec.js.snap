exports[`Home Page > toMatchSnapshot - Body #0`] = `
<body class="mat-typography">
  <app-root _nghost-vih-c124="" ng-version="10.0.7"
    ><mat-toolbar
      _ngcontent-vih-c124=""
      color="primary"
      class="mat-toolbar header mat-primary mat-toolbar-single-row"
      ng-reflect-color="primary"
      ><span _ngcontent-vih-c124="" class="title-center mat-h1"
        >Welcome to angular10-reactive-forms!</span
      ></mat-toolbar
    ><mat-sidenav-container
      _ngcontent-vih-c124=""
      class="mat-drawer-container mat-sidenav-container"
      ><div class="mat-drawer-backdrop ng-star-inserted"></div>
      <!--bindings={
  "ng-reflect-ng-if": "true"
}--><mat-sidenav-content
        _ngcontent-vih-c124=""
        class="mat-drawer-content mat-sidenav-content"
        ><form
          _ngcontent-vih-c124=""
          novalidate=""
          class="example-form ng-untouched ng-pristine ng-invalid"
          ng-reflect-form="[object Object]"
        >
          <mat-form-field
            _ngcontent-vih-c124=""
            class="mat-form-field example-full-width ng-tns-c74-0 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder ng-untouched ng-pristine ng-invalid"
            ><div class="mat-form-field-wrapper ng-tns-c74-0">
              <div class="mat-form-field-flex ng-tns-c74-0">
                <!--bindings={
  "ng-reflect-ng-if": "false"
}--><!--bindings={
  "ng-reflect-ng-if": "0"
}-->
                <div class="mat-form-field-infix ng-tns-c74-0">
                  <input
                    _ngcontent-vih-c124=""
                    matinput=""
                    formcontrolname="name"
                    data-cy="nameInput"
                    class="mat-input-element mat-form-field-autofill-control ng-tns-c74-0 ng-untouched ng-pristine ng-invalid cdk-text-field-autofill-monitored"
                    ng-reflect-placeholder="Name"
                    ng-reflect-name="name"
                    ng-reflect-error-state-matcher="[object Object]"
                    id="mat-input-0"
                    data-placeholder="Name"
                    aria-invalid="false"
                    aria-required="false"
                  /><span class="mat-form-field-label-wrapper ng-tns-c74-0"
                    ><label
                      class="mat-form-field-label ng-tns-c74-0 mat-empty mat-form-field-empty ng-star-inserted"
                      ng-reflect-disabled="true"
                      id="mat-form-field-label-1"
                      ng-reflect-ng-switch="true"
                      for="mat-input-0"
                      aria-owns="mat-input-0"
                      ><!--bindings={
  "ng-reflect-ng-switch-case": "false"
}--><mat-label _ngcontent-vih-c124="" class="ng-tns-c74-0 ng-star-inserted"
                        >Name</mat-label
                      ><!--bindings={
  "ng-reflect-ng-switch-case": "true"
}--><!--bindings={
  "ng-reflect-ng-if": "false"
}--></label
                    ><!--bindings={
  "ng-reflect-ng-if": "true"
}--></span
                  >
                </div>
                <!--bindings={
  "ng-reflect-ng-if": "0"
}-->
              </div>
              <div
                class="mat-form-field-underline ng-tns-c74-0 ng-star-inserted"
              >
                <span class="mat-form-field-ripple ng-tns-c74-0"></span>
              </div>
              <!--bindings={
  "ng-reflect-ng-if": "true"
}-->
              <div
                class="mat-form-field-subscript-wrapper ng-tns-c74-0"
                ng-reflect-ng-switch="hint"
              >
                <!--bindings={
  "ng-reflect-ng-switch-case": "error"
}-->
                <div
                  class="mat-form-field-hint-wrapper ng-tns-c74-0 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                  style="opacity: 1; transform: translateY(0%);"
                >
                  <!--bindings={
  "ng-reflect-ng-if": ""
}-->
                  <div class="mat-form-field-hint-spacer ng-tns-c74-0"></div>
                </div>
                <!--bindings={
  "ng-reflect-ng-switch-case": "hint"
}-->
              </div>
            </div></mat-form-field
          ><mat-form-field
            _ngcontent-vih-c124=""
            class="mat-form-field example-full-width ng-tns-c74-1 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder ng-untouched ng-pristine ng-invalid"
            ><div class="mat-form-field-wrapper ng-tns-c74-1">
              <div class="mat-form-field-flex ng-tns-c74-1">
                <!--bindings={
  "ng-reflect-ng-if": "false"
}--><!--bindings={
  "ng-reflect-ng-if": "0"
}-->
                <div class="mat-form-field-infix ng-tns-c74-1">
                  <input
                    _ngcontent-vih-c124=""
                    matinput=""
                    formcontrolname="email"
                    data-cy="emailInput"
                    class="mat-input-element mat-form-field-autofill-control ng-tns-c74-1 ng-untouched ng-pristine ng-invalid cdk-text-field-autofill-monitored"
                    ng-reflect-placeholder="Ex. pat@example.com"
                    ng-reflect-name="email"
                    ng-reflect-error-state-matcher="[object Object]"
                    id="mat-input-1"
                    data-placeholder="Ex. pat@example.com"
                    aria-invalid="false"
                    aria-required="false"
                  /><span class="mat-form-field-label-wrapper ng-tns-c74-1"
                    ><label
                      class="mat-form-field-label ng-tns-c74-1 mat-empty mat-form-field-empty ng-star-inserted"
                      ng-reflect-disabled="true"
                      id="mat-form-field-label-3"
                      ng-reflect-ng-switch="true"
                      for="mat-input-1"
                      aria-owns="mat-input-1"
                      ><!--bindings={
  "ng-reflect-ng-switch-case": "false"
}--><mat-label _ngcontent-vih-c124="" class="ng-tns-c74-1 ng-star-inserted"
                        >Email</mat-label
                      ><!--bindings={
  "ng-reflect-ng-switch-case": "true"
}--><!--bindings={
  "ng-reflect-ng-if": "false"
}--></label
                    ><!--bindings={
  "ng-reflect-ng-if": "true"
}--></span
                  >
                </div>
                <!--bindings={
  "ng-reflect-ng-if": "0"
}-->
              </div>
              <div
                class="mat-form-field-underline ng-tns-c74-1 ng-star-inserted"
              >
                <span class="mat-form-field-ripple ng-tns-c74-1"></span>
              </div>
              <!--bindings={
  "ng-reflect-ng-if": "true"
}-->
              <div
                class="mat-form-field-subscript-wrapper ng-tns-c74-1"
                ng-reflect-ng-switch="hint"
              >
                <!--bindings={
  "ng-reflect-ng-switch-case": "error"
}-->
                <div
                  class="mat-form-field-hint-wrapper ng-tns-c74-1 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                  style="opacity: 1; transform: translateY(0%);"
                >
                  <!--bindings={
  "ng-reflect-ng-if": ""
}-->
                  <div class="mat-form-field-hint-spacer ng-tns-c74-1"></div>
                </div>
                <!--bindings={
  "ng-reflect-ng-switch-case": "hint"
}-->
              </div>
            </div></mat-form-field
          >
          <div _ngcontent-vih-c124="" class="misc-bottom-padding">
            <mat-label _ngcontent-vih-c124="">Gender:</mat-label
            ><mat-radio-group
              _ngcontent-vih-c124=""
              role="radiogroup"
              aria-label="Select an option"
              formcontrolname="gender"
              class="mat-radio-group example-radio-group ng-untouched ng-pristine ng-valid"
              ng-reflect-name="gender"
              ><mat-radio-button
                _ngcontent-vih-c124=""
                class="mat-radio-button example-radio-button mat-radio-checked mat-accent ng-star-inserted"
                ng-reflect-value="Male"
                tabindex="-1"
                id="mat-radio-2"
                ><label class="mat-radio-label" for="mat-radio-2-input"
                  ><div class="mat-radio-container">
                    <div class="mat-radio-outer-circle"></div>
                    <div class="mat-radio-inner-circle"></div>
                    <input
                      type="radio"
                      class="mat-radio-input cdk-visually-hidden"
                      id="mat-radio-2-input"
                      tabindex="0"
                      name="mat-radio-group-0"
                      value="Male"
                    />
                    <div
                      mat-ripple=""
                      class="mat-ripple mat-radio-ripple mat-focus-indicator"
                      ng-reflect-trigger="[object HTMLLabelElement]"
                      ng-reflect-disabled="false"
                      ng-reflect-centered="true"
                      ng-reflect-radius="20"
                      ng-reflect-animation="[object Object]"
                    >
                      <div
                        class="mat-ripple-element mat-radio-persistent-ripple"
                      ></div>
                    </div>
                  </div>
                  <div class="mat-radio-label-content">
                    <span style="display: none;">&nbsp;</span>Male
                  </div></label
                ></mat-radio-button
              ><mat-radio-button
                _ngcontent-vih-c124=""
                class="mat-radio-button example-radio-button mat-accent ng-star-inserted"
                ng-reflect-value="Female"
                tabindex="-1"
                id="mat-radio-3"
                ><label class="mat-radio-label" for="mat-radio-3-input"
                  ><div class="mat-radio-container">
                    <div class="mat-radio-outer-circle"></div>
                    <div class="mat-radio-inner-circle"></div>
                    <input
                      type="radio"
                      class="mat-radio-input cdk-visually-hidden"
                      id="mat-radio-3-input"
                      tabindex="0"
                      name="mat-radio-group-0"
                      value="Female"
                    />
                    <div
                      mat-ripple=""
                      class="mat-ripple mat-radio-ripple mat-focus-indicator"
                      ng-reflect-trigger="[object HTMLLabelElement]"
                      ng-reflect-disabled="false"
                      ng-reflect-centered="true"
                      ng-reflect-radius="20"
                      ng-reflect-animation="[object Object]"
                    >
                      <div
                        class="mat-ripple-element mat-radio-persistent-ripple"
                      ></div>
                    </div>
                  </div>
                  <div class="mat-radio-label-content">
                    <span style="display: none;">&nbsp;</span>Female
                  </div></label
                ></mat-radio-button
              ><!--bindings={
  "ng-reflect-ng-for-of": "Male,Female"
}--></mat-radio-group
            >
          </div>
          <mat-form-field
            _ngcontent-vih-c124=""
            class="mat-form-field ng-tns-c74-2 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder ng-untouched ng-pristine ng-invalid"
            ><div class="mat-form-field-wrapper ng-tns-c74-2">
              <div class="mat-form-field-flex ng-tns-c74-2">
                <!--bindings={
  "ng-reflect-ng-if": "false"
}--><!--bindings={
  "ng-reflect-ng-if": "0"
}-->
                <div class="mat-form-field-infix ng-tns-c74-2">
                  <input
                    _ngcontent-vih-c124=""
                    matinput=""
                    readonly="true"
                    formcontrolname="dob"
                    class="mat-input-element mat-form-field-autofill-control ng-tns-c74-2 ng-untouched ng-pristine ng-invalid cdk-text-field-autofill-monitored"
                    ng-reflect-readonly=""
                    ng-reflect-placeholder="Date of birth"
                    ng-reflect-name="dob"
                    ng-reflect-mat-datepicker="[object Object]"
                    id="mat-input-2"
                    data-placeholder="Date of birth"
                    aria-invalid="false"
                    aria-required="false"
                    aria-haspopup="dialog"
                  /><mat-datepicker
                    _ngcontent-vih-c124=""
                    class="ng-tns-c74-2"
                  ></mat-datepicker
                  ><!--container--><span
                    class="mat-form-field-label-wrapper ng-tns-c74-2"
                    ><label
                      class="mat-form-field-label ng-tns-c74-2 mat-empty mat-form-field-empty ng-star-inserted"
                      ng-reflect-disabled="true"
                      id="mat-form-field-label-5"
                      ng-reflect-ng-switch="false"
                      for="mat-input-2"
                      aria-owns="mat-input-2"
                      ><span class="ng-tns-c74-2 ng-star-inserted"
                        >Date of birth</span
                      ><!--ng-container--><!--bindings={
  "ng-reflect-ng-switch-case": "false"
}--><!--bindings={
  "ng-reflect-ng-switch-case": "true"
}--><!--bindings={
  "ng-reflect-ng-if": "false"
}--></label
                    ><!--bindings={
  "ng-reflect-ng-if": "true"
}--></span
                  >
                </div>
                <div
                  class="mat-form-field-suffix ng-tns-c74-2 ng-star-inserted"
                >
                  <mat-datepicker-toggle
                    _ngcontent-vih-c124=""
                    matsuffix=""
                    class="mat-datepicker-toggle ng-tns-c74-2"
                    ng-reflect-datepicker="[object Object]"
                    tabindex="-1"
                    ><button
                      mat-icon-button=""
                      type="button"
                      class="mat-focus-indicator mat-icon-button mat-button-base"
                      ng-reflect-disabled="false"
                      aria-haspopup="dialog"
                      aria-label="Open calendar"
                      tabindex="0"
                    >
                      <span class="mat-button-wrapper"
                        ><svg
                          viewBox="0 0 24 24"
                          width="24px"
                          height="24px"
                          fill="currentColor"
                          focusable="false"
                          class="mat-datepicker-toggle-default-icon ng-star-inserted"
                        >
                          <path
                            d="M19 3h-1V1h-2v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V8h14v11zM7 10h5v5H7z"
                          ></path></svg
                        ><!--bindings={
  "ng-reflect-ng-if": "true"
}--></span
                      >
                      <div
                        matripple=""
                        class="mat-ripple mat-button-ripple mat-button-ripple-round"
                        ng-reflect-disabled="false"
                        ng-reflect-centered="true"
                        ng-reflect-trigger="[object HTMLButtonElement]"
                      ></div>
                      <div class="mat-button-focus-overlay"></div></button
                  ></mat-datepicker-toggle>
                </div>
                <!--bindings={
  "ng-reflect-ng-if": "1"
}-->
              </div>
              <div
                class="mat-form-field-underline ng-tns-c74-2 ng-star-inserted"
              >
                <span class="mat-form-field-ripple ng-tns-c74-2"></span>
              </div>
              <!--bindings={
  "ng-reflect-ng-if": "true"
}-->
              <div
                class="mat-form-field-subscript-wrapper ng-tns-c74-2"
                ng-reflect-ng-switch="hint"
              >
                <!--bindings={
  "ng-reflect-ng-switch-case": "error"
}-->
                <div
                  class="mat-form-field-hint-wrapper ng-tns-c74-2 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                  style="opacity: 1; transform: translateY(0%);"
                >
                  <!--bindings={
  "ng-reflect-ng-if": ""
}-->
                  <div class="mat-form-field-hint-spacer ng-tns-c74-2"></div>
                </div>
                <!--bindings={
  "ng-reflect-ng-switch-case": "hint"
}-->
              </div>
            </div></mat-form-field
          ><mat-form-field
            _ngcontent-vih-c124=""
            class="mat-form-field ng-tns-c74-3 mat-primary mat-form-field-type-mat-select mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder ng-untouched ng-pristine ng-valid"
            ><div class="mat-form-field-wrapper ng-tns-c74-3">
              <div class="mat-form-field-flex ng-tns-c74-3">
                <!--bindings={
  "ng-reflect-ng-if": "false"
}--><!--bindings={
  "ng-reflect-ng-if": "0"
}-->
                <div class="mat-form-field-infix ng-tns-c74-3">
                  <mat-select
                    _ngcontent-vih-c124=""
                    role="listbox"
                    formcontrolname="grade"
                    data-cy="select"
                    class="mat-select ng-tns-c116-4 ng-tns-c74-3 mat-select-empty ng-untouched ng-pristine ng-valid ng-star-inserted"
                    ng-reflect-name="grade"
                    id="mat-select-0"
                    tabindex="0"
                    aria-labelledby="mat-form-field-label-7"
                    aria-required="false"
                    aria-disabled="false"
                    aria-invalid="false"
                    aria-multiselectable="false"
                    ><div
                      cdk-overlay-origin=""
                      aria-hidden="true"
                      class="mat-select-trigger ng-tns-c116-4"
                    >
                      <div
                        class="mat-select-value ng-tns-c116-4"
                        ng-reflect-ng-switch="true"
                      >
                        <span
                          class="mat-select-placeholder ng-tns-c116-4 ng-star-inserted"
                          >&nbsp;</span
                        ><!--bindings={
  "ng-reflect-ng-switch-case": "true"
}--><!--bindings={
  "ng-reflect-ng-switch-case": "false"
}-->
                      </div>
                      <div class="mat-select-arrow-wrapper ng-tns-c116-4">
                        <div class="mat-select-arrow ng-tns-c116-4"></div>
                      </div>
                    </div>
                    <!--bindings={
  "ng-reflect-offset-y": "0"
}--></mat-select
                  ><span class="mat-form-field-label-wrapper ng-tns-c74-3"
                    ><label
                      class="mat-form-field-label ng-tns-c74-3 mat-empty mat-form-field-empty ng-star-inserted"
                      ng-reflect-disabled="true"
                      id="mat-form-field-label-7"
                      ng-reflect-ng-switch="true"
                      for="mat-select-0"
                      aria-owns="mat-select-0"
                      ><!--bindings={
  "ng-reflect-ng-switch-case": "false"
}--><mat-label _ngcontent-vih-c124="" class="ng-tns-c74-3 ng-star-inserted"
                        >Grade</mat-label
                      ><!--bindings={
  "ng-reflect-ng-switch-case": "true"
}--><!--bindings={
  "ng-reflect-ng-if": "false"
}--></label
                    ><!--bindings={
  "ng-reflect-ng-if": "true"
}--></span
                  >
                </div>
                <!--bindings={
  "ng-reflect-ng-if": "0"
}-->
              </div>
              <div
                class="mat-form-field-underline ng-tns-c74-3 ng-star-inserted"
              >
                <span class="mat-form-field-ripple ng-tns-c74-3"></span>
              </div>
              <!--bindings={
  "ng-reflect-ng-if": "true"
}-->
              <div
                class="mat-form-field-subscript-wrapper ng-tns-c74-3"
                ng-reflect-ng-switch="hint"
              >
                <!--bindings={
  "ng-reflect-ng-switch-case": "error"
}-->
                <div
                  class="mat-form-field-hint-wrapper ng-tns-c74-3 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                  style="opacity: 1; transform: translateY(0%);"
                >
                  <!--bindings={
  "ng-reflect-ng-if": ""
}-->
                  <div class="mat-form-field-hint-spacer ng-tns-c74-3"></div>
                </div>
                <!--bindings={
  "ng-reflect-ng-switch-case": "hint"
}-->
              </div>
            </div></mat-form-field
          ><mat-form-field
            _ngcontent-vih-c124=""
            class="mat-form-field multiple-items ng-tns-c74-5 mat-primary mat-form-field-type-mat-chip-list mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder"
            ><div class="mat-form-field-wrapper ng-tns-c74-5">
              <div class="mat-form-field-flex ng-tns-c74-5">
                <!--bindings={
  "ng-reflect-ng-if": "false"
}--><!--bindings={
  "ng-reflect-ng-if": "0"
}-->
                <div class="mat-form-field-infix ng-tns-c74-5">
                  <mat-chip-list
                    _ngcontent-vih-c124=""
                    class="mat-chip-list ng-tns-c74-5"
                    id="mat-chip-list-0"
                    tabindex="-1"
                    aria-disabled="false"
                    aria-invalid="false"
                    aria-multiselectable="false"
                    aria-orientation="horizontal"
                    ><div class="mat-chip-list-wrapper">
                      <!--bindings={
  "ng-reflect-ng-for-of": ""
}--><input
                        _ngcontent-vih-c124=""
                        placeholder="Add subjects"
                        data-cy="subjectInput"
                        class="mat-chip-input mat-input-element"
                        ng-reflect-placeholder="Add subjects"
                        ng-reflect-chip-list="[object Object]"
                        ng-reflect-separator-key-codes="13,188"
                        ng-reflect-add-on-blur="true"
                        id="mat-chip-list-input-0"
                      /></div></mat-chip-list
                  ><span class="mat-form-field-label-wrapper ng-tns-c74-5"
                    ><label
                      class="mat-form-field-label ng-tns-c74-5 mat-empty mat-form-field-empty ng-star-inserted"
                      ng-reflect-disabled="true"
                      id="mat-form-field-label-9"
                      ng-reflect-ng-switch="false"
                      for="mat-chip-list-input-0"
                      aria-owns="mat-chip-list-input-0"
                      ><span class="ng-tns-c74-5 ng-star-inserted"
                        >Add subjects</span
                      ><!--ng-container--><!--bindings={
  "ng-reflect-ng-switch-case": "false"
}--><!--bindings={
  "ng-reflect-ng-switch-case": "true"
}--><!--bindings={
  "ng-reflect-ng-if": "false"
}--></label
                    ><!--bindings={
  "ng-reflect-ng-if": "true"
}--></span
                  >
                </div>
                <!--bindings={
  "ng-reflect-ng-if": "0"
}-->
              </div>
              <div
                class="mat-form-field-underline ng-tns-c74-5 ng-star-inserted"
              >
                <span class="mat-form-field-ripple ng-tns-c74-5"></span>
              </div>
              <!--bindings={
  "ng-reflect-ng-if": "true"
}-->
              <div
                class="mat-form-field-subscript-wrapper ng-tns-c74-5"
                ng-reflect-ng-switch="hint"
              >
                <!--bindings={
  "ng-reflect-ng-switch-case": "error"
}-->
                <div
                  class="mat-form-field-hint-wrapper ng-tns-c74-5 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                  style="opacity: 1; transform: translateY(0%);"
                >
                  <!--bindings={
  "ng-reflect-ng-if": ""
}-->
                  <div class="mat-form-field-hint-spacer ng-tns-c74-5"></div>
                </div>
                <!--bindings={
  "ng-reflect-ng-switch-case": "hint"
}-->
              </div>
            </div></mat-form-field
          >
          <div _ngcontent-vih-c124="" class="button-wrapper">
            <button
              _ngcontent-vih-c124=""
              mat-flat-button=""
              color="primary"
              data-cy="submit"
              class="mat-focus-indicator mat-flat-button mat-button-base mat-primary"
              ng-reflect-color="primary"
            >
              <span class="mat-button-wrapper">Submit</span>
              <div
                matripple=""
                class="mat-ripple mat-button-ripple"
                ng-reflect-disabled="false"
                ng-reflect-centered="false"
                ng-reflect-trigger="[object HTMLButtonElement]"
              ></div>
              <div class="mat-button-focus-overlay"></div>
            </button>
          </div></form></mat-sidenav-content
      ><!--bindings={
  "ng-reflect-ng-if": "false"
}--></mat-sidenav-container
    ></app-root
  >
  <script src="runtime.js" defer=""></script>
  <script src="polyfills.js" defer=""></script>
  <script src="styles.js" defer=""></script>
  <script src="vendor.js" defer=""></script>
  <script src="main.js" defer=""></script>

  <div
    class="cdk-live-announcer-element cdk-visually-hidden"
    aria-atomic="true"
    aria-live="polite"
  ></div>
</body>
`;
